velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -184,6 +201,78 @@ void WindowFuzzer::go(const std::string& /*planPath*/) {
   VELOX_NYI();
 }
 
+void WindowFuzzer::testAlternativePlans(
+    const std::vector<std::string>& partitionKeys,
+    const std::vector<SortingKeyAndOrder>& sortingKeysAndOrders,
+    const std::string& frame,
+    const std::string& functionCall,
+    const std::vector<RowVectorPtr>& input,
+    bool customVerification,
+    const velox::test::ResultOrError& expected) {
+  std::vector<AggregationFuzzerBase::PlanWithSplits> plans;
+
+  std::vector<std::string> allKeys;
+  for (const auto& key : partitionKeys) {
+    allKeys.push_back(key + " NULLS FIRST");
+  }
+  for (const auto& keyAndOrder : sortingKeysAndOrders) {
+    allKeys.push_back(folly::to<std::string>(
+        keyAndOrder.key_,
+        " ",
+        keyAndOrder.order_,
+        " ",
+        keyAndOrder.nullsOrder_));
+  }
+
+  // Streaming window from values.
+  if (!allKeys.empty()) {
+    plans.push_back(
+        {PlanBuilder()
+             .values(input)
+             .orderBy(allKeys, false)
+             .streamingWindow(
+                 {fmt::format("{} over ({})", functionCall, frame)})
+             .planNode(),
+         {}});
+  }
+
+  // With TableScan.
+  auto directory = exec::test::TempDirectoryPath::create();
+  const auto inputRowType = asRowType(input[0]->type());
+  if (isTableScanSupported(inputRowType)) {
+    auto splits = makeSplits(input, directory->path);
+
+    plans.push_back(
+        {PlanBuilder()
+             .tableScan(inputRowType)
+             .localPartition(partitionKeys)
+             .window({fmt::format("{} over ({})", functionCall, frame)})
+             .planNode(),
+         splits});
+
+    if (!allKeys.empty()) {
+      plans.push_back(
+          {PlanBuilder()
+               .tableScan(inputRowType)
+               .orderBy(allKeys, false)
+               .streamingWindow(
+                   {fmt::format("{} over ({})", functionCall, frame)})
+               .planNode(),
+           splits});
+    }
+  }
+
+  for (const auto& plan : plans) {
+    testPlan(
```

Comment:
nit : we have testPlans, which also supports with spilling, is there a reason we cant use that ?

Reply:
+1. Its important we test spilling as well for windows. We should generate plans with that.

Reply:
Thank you for the suggestion. I'll add spilling plans in a follow-up PR.

Reply:
> nit : we have testPlans, which also supports with spilling, is there a reason we cant use that ?

testPlans() is in AggregationFuzzer and has aggregation-specific code (i.e., hasPartialGroupBy()). So WindowFuzzer needs to have its own "testPlans" method.